### PR TITLE
fixes #145, add a GH action to check that watchdog code in Jenkinsfile is not bei…

### LINKED ISSUE #145: Create GitHub workflow to act as PR watchdog; protect the release pipelines
In order to both better protect `snapshot`, `release`, and maybe `master`, we should create a .travis.yml that

* checks to make sure that Jenkinsfile still has the verbatim watchdog block
* performs similar checks to the current watchdog
* does additional checks that should be easy in back or python, such as ensuring that Zenodo and AWS targets (all publication targets) do not seem coherent or do not exist; possibly enforce a rule such as "null"

### PR DESCRIPTION
**Pull request by sierra-moxon**
…ng changed on PR

I did this because I have to start editing a test pipeline and I want to protect myself from doing damage to any PROD system.

fixes #145 by doing a diff between the PR'd branch (triggered on PR only) and master to see if it can find any `git diff` output in the watchdog function/code block.

if it finds a change within the watchdog function/code block, it will return an error status for this action report: `ERROR: The watchdog function in the Jenkinsfile has been modified.`
else, it will succeed and print `watchdog OK`.  Going one step further and enabling the branch protection rule that all PRs must pass checks would cement this as a failure status for any changes to the Jenkins file.

to circumvent the action, we can do one of two things: 
1) turn off branch protections again - so we can merge even with a failing check
2) just set the github action to no longer run on PR

### CODE REVIEW

File: .github/workflows/check-watchdog.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      uses: actions/checkout@v2
+      with:
+        ref: ${{ github.head_ref }}
+
+    - name: Fetch master branch
+      run: git fetch origin master:master
+
+    - name: Check for modifications in the watchdog function
+      run: |
+        set -x
```
Comment (sierra-moxon) [inline .github/workflows/check-watchdog.yaml:22]: note: this is just helpful in debugging the shell command if things go wrong.

File: .github/workflows/check-watchdog.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        if git diff --unified=0 master ${{ github.head_ref }} Jenkinsfile | grep 'void watchdog()'; then
+          echo "ERROR: The watchdog function in the Jenkinsfile has been modified."
```
Comment (kltm) [inline .github/workflows/check-watchdog.yaml:24]: This may be better phrased with the addition of: "Who let the dogs out? Who, who, who, who, who?"

**Review by kltm — APPROVED** [review]
Looks good as a first check and as a launching place for future safety additions.

**Comment by kltm** [comment]
This will automatically get "tested" when next scanned.

**Pull request merged**